src/include/migraphx/op/random_uniform.hpp

```diff
@@ -75,7 +75,16 @@ struct random_uniform
 
         result.visit([&](auto output) {
             using type = typename decltype(output)::value_type;
-            if constexpr(std::is_integral<type>{})
+            if constexpr(
+                std::is_integral_v<type>
```

Comment:
Why do we need to instantiate the static check? That is a compile-time property check of the static-if and not a runtime check. `std::is_integral_v<>` evaluates to `std::integral<>::value`, which is `true` for integral types and `false` for others. Honestly, I have not seen someone using static checks that way.

Reply:
> Why do we need to instantiate the static check?

You are instantiating the template in both cases.

1. By constructing it we get an `integral_constant` instead of bool which is more powerful in some cases(such as passing this to an intermediate function).
2. Not all type traits provide variable templates, so we need to write it like that
3. Consistency. We shouldnt need to think about whether we will pass it to a function or whether it supports a variable template. We write it the same way everytime.